src/Compiler/Microsoft.CodeAnalysis.Razor/src/SymbolExtensions.cs

```diff
@@ -14,6 +14,23 @@ internal static class SymbolExtensions
 
     internal static bool HasFullName(this ISymbol symbol, string fullName)
     {
+        // generally we expect this call to fail, so try and match on the simple name first, before
+        // checking the fully qualified name, to prevent unnecesary string allocation from ToDisplayString 
+        if (symbol.Name.Length > fullName.Length)
+        {
+            return false;
+        }
+
+        // the simple name doesn't include generic type params, so skip over them in the full name if they are present
+        int fullNameEndIndex = fullName[^1] == '>' ? fullName.LastIndexOf('<') : fullName.Length;
+        int fullNameStartIndex = fullName.LastIndexOf('.', fullNameEndIndex - 1) + 1;
```

Comment:
❔ Are nested types separated by `.` or `+`?

Reply:
`+`. Same reply as to Davids comment above, this isn't general purpose and only needs to work for a limited set of types we control.